cmd/gateclient/oauth.go

```diff
@@ -0,0 +1,85 @@
+package gateclient
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"os/exec"
+	"runtime"
+
+	"golang.org/x/oauth2"
+)
+
+const localWebServer = "localhost:8085"
+
+func startWebServer() (codeCh chan string, err error) {
```

Comment:
Can you add a test for this please? - see: https://pkg.go.dev/net/http/httptest
We can validate:
1. `code` FormValue is supplied in test and plumbed through to codeCh
2. http response recorder has correct output
3. listener closed
4. `error` FormValue (per next comment - if part of this PR).